buildSrc/src/main/java/org/elasticsearch/gradle/Version.java

```diff
@@ -16,35 +16,43 @@
     /**
      * Suffix on the version name.
      */
-    private final String suffix;
+    private final String qualifier;
 
     private static final Pattern pattern =
             Pattern.compile("(\\d)+\\.(\\d+)\\.(\\d+)(-alpha\\d+|-beta\\d+|-rc\\d+)?(-SNAPSHOT)?");
 
-    public Version(int major, int minor, int revision, String suffix, boolean snapshot) {
+    Version(int major, int minor, int revision, String qualifier, boolean snapshot) {
         Objects.requireNonNull(major, "major version can't be null");
         Objects.requireNonNull(minor, "minor version can't be null");
         Objects.requireNonNull(revision, "revision version can't be null");
         this.major = major;
         this.minor = minor;
         this.revision = revision;
         this.snapshot = snapshot;
-        this.suffix = suffix == null ? "" : suffix;
+        if (qualifier == null || qualifier.isEmpty()) {
+            this.qualifier = "";
+        } else if (qualifier.startsWith("-")) {
+            this.qualifier = qualifier;
+        } else {
+            this.qualifier = "-" + qualifier;
+        }
 
         int suffixOffset = 0;
-        if (this.suffix.isEmpty()) {
-            // no suffix will be considered smaller, uncomment to change that
+        if (this.qualifier.isEmpty()) {
+            // no qualifier will be considered smaller, uncomment to change that
             // suffixOffset = 100;
```

Comment:
What is the purpose of having this commented out conditional block? With the new rules, we shouldn't ever be comparing a qualified build against each other or a non qualified build, since qualified and non qualified of the same version have the exact same bwc behavior.

Reply:
I added this comment some time ago when I discovered that `7.0.0` would be considered smaller than `7.0.0-alpha1` which is incorrect, but changing that caused failures in `VersionCollection` and I didn't want to dig deeper at that time so just left this comment as a reminder. 
The qualifier will not be relevant indeed for bwc behavior, but it's still relevant for the build, e.x. we still want to use an beta over an alpha if available so the build still has to order them.
I'm hoping that with the metadata format in  #32872 the proper ordering will be clearer and the implementation will clean up as a result as well.
I think these are simple things, yet hard to get right with harsh consequences if they are not right, I'm reluctant to change them without the visualization described in #32872 to be able to better reason about it.

Reply:
> we still want to use an beta over an alpha

There shouldn't be anything needing to choose a beta over an alpha? There should be nothing using any qualified build to check bwc.

Reply:
I removed the commented code to avoid confusion. 
To make sure my understanding is correct: we will need to consider qualifier in the build, i.e. so we don't BWC test against an alpha if a beta is already available. This is for build only, server will not know of qualifier, so it has no way to decide based on it.

Reply:
I don't know what you mean by "consider qualifier in the build". There should be no attempt to test bwc of any qualified version, so I don't believe the build needs to know about it in our build Version, since that class is all about which versions we bwc test against.

Reply:
The version class is now used more broadly,  including in figuring out dependencies ( as it's string value is set as project.version which is then used when considering dependencies ).
My understanding was that in BWC testing we _will_ want to pic up `alpha`, `beta` etc as these become available, I'm starting to realize that since we don't offer guarantees we won't have to test with those and would just test against the latest snapshot and be ok. 

I don't think we should remove the qualifier from the version right now, we should eventually better express requirements for a version used in bwc, perhaps as part of #32872.